src/header/mod.rs

```diff
@@ -106,6 +112,86 @@ pub struct Header {
 pub struct Vlrs<'a>(Chain<Iter<'a, Vlr>, Iter<'a, Vlr>>);
 
 impl Header {
+    /// Reads all header, vlr and evlr data from file and returns the complete header
+    pub fn read_and_build_from<R: Read + Seek>(mut read: R) -> Result<Self> {
```

Comment:
In this repo, we've used `new` to build structure from a `Read`:

```suggestion
    pub fn new<R: Read + Seek>(mut read: R) -> Result<Self> {
```